feat(icons): add new icons package

jharvey10 opened the pull request:
- web-app depends on icons now
- removes manual requirement to rebuild icons and commit generated components to the repo
- assorted updates to micromanage link

jharvey10 committed:
feat(icons): add new icons package

- web-app depends on icons now
- removes manual requirement to rebuild icons and commit generated components to the repo
- assorted updates to micromanage link

jharvey10 committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into package/icons

jharvey10 committed:
build: remove duplicate invocation of micromanage link

mattrosno reviewed (commented):
If you do root-level `npm install` then step into the web app and `npm run dev`, you get this error, probably because the icons package hasn't been built yet.

`Module not found: Can't resolve '@carbon-platform/icons'`

Second attempt, I ran `npm run all:build` at root and then in web app directory `npm run dev` again, which worked.

@jdharvey-ibm how do we ensure that the icons package has been built before doing anything in the web app, since there is that dependency?

jharvey10 committed:
build: update package-lock.json

jharvey10 committed:
fix(icons): add module key to package.json

- add auto-build of packages to "prepare" script after husky install

jharvey10 committed:
chore(release): bump icons to 0.0.2

mattrosno reviewed (dismissed):
Approved - just a reminder to add docs on build order prior to develop.

jharvey10 committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into package/icons

jharvey10 committed:
build(micromange): auto update local package deps during re-version

- turn on eslint max-len for comments to prevent long lines prettier doesn't fix
- destructure imports for better code analysis
- add some function documentation
- improve the way in which updated packages are discovered
- exhaustively find all packages that need updates, including because of dependent package changes

jharvey10 committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into package/icons

jharvey10 committed:
ci: ignore scripts from code coverage reports
